### ISSUE TITLE
PubSub ingester chokes on null value in payload

### ISSUE THREAD

**Issue by jwhitlock** [mozilla-it/ctms-api#354]
This line of the pubsub ingester:

https://github.com/mozilla-it/ctms-api/blob/904a8e40fc3e312dabc769d0742e6668d18f4be4/ctms/app.py#L1030

is raising an exception:

```python
TypeError("argument of type 'NoneType' is not iterable")
```

It is expecting a payload dictionary like:

```json
{
   "firestore_table:stripe_id": {
      "object": "stripe_type",
      "other": "values"
   }
}
```

This suggests a payload dictionary like:

```json
{
   "obj:obj_id": null
}
```

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/CTMS-126)

**Comment by jwhitlock** [comment]
On the CTMS side, something like this may work:

```python
    if hasattr(payload, "keys"):
        if all(key.count(":") == 1 for key in payload.keys()):
            # Item dictionary, "firebase_table:id" -> Stripe object
            items = []
            for key, value in payload.items():
                if value:
                    items.append(value)
                else:
                    logger.warning("PubSub key %s had empty value %s", key, value)
        else:
            items = [payload]  # One Stripe object, or maybe unknown dictionary
```

Alternatively, the pubsub cloud function could discard null values.

We may need to coordinate with the FxA team to determine why null values are being added to Firestore.

**Comment by bsieber-mozilla** [comment]
Thanks for looking into and outlining this issue; I've implemented your proposed solution in PR #355 

I think keeping an eye on the amount of `None` we're receiving could be insightful. It looks like there was only one triggered incident; one message being held up.

**Comment by bsieber-mozilla** [comment]
I believe we can close this now.
